### PR TITLE
Update FirstBank

### PR DESCRIPTION
**Pull request by smmelot-16**
Adding 2FA methods currently available.

### CODE REVIEW

File: entries/e/efirstbank.com.json [hunk 1]
```diff
@@ -2,11 +2,17 @@
...
-    },
+    "tfa": [
+      "custom-software",
+      "custom-hardware"
+    ],
+    "custom-software": [
+      "FirstBank App Push Notification"
+    ],
+    "custom-hardware": [
+      "Only available for commerical users"
```
Comment (Carlgo11) [inline entries/e/efirstbank.com.json:13]: 1. The value here should be the name of the hardware used. Any restrictions can be displayed as a note.
2. I don't see any documentation for hardware 2FA on efirstbank.com.

File: entries/e/efirstbank.com.json [hunk 2]
```diff
@@ -2,11 +2,17 @@
...
-      "email": "banking@efirstbank.com",
-      "facebook": "efirstbank",
-      "twitter": "efirstbank"
-    },
+    "tfa": [
+      "custom-software",
+      "custom-hardware"
+    ],
+    "custom-software": [
+      "FirstBank App Push Notification"
```
Comment (Carlgo11) [inline entries/e/efirstbank.com.json]:
```suggestion
      "FirstBank App"
```

**Review by Carlgo11 — CHANGES_REQUESTED** [review]
I have some questions about the implementation:
1. There's talk about Quick Login and OTP codes. Are these seperate ways of authentication?
1. The documentation page says _either security questions or a one time password is used on new devices._ If the 2FA prompt can be bypassed by security questions it wouldn't qualify here.

**Comment by Carlgo11** [comment]
As the requested changes haven't been resolved yet I'm going to label this PR as inactive.
Unless there's activity within the next week I'll close the PR.

File: entries/e/efirstbank.com.json (hunk 1 above)
Comment (smmelot-16) [reply entries/e/efirstbank.com.json:13]: Apologies, I had to go try and track down some of this information. I couldn't find any specific information about which hardware token is being used, other than the website just mentioning that tokens are used: https://efirstbank.com/customer-service/questions-answers.htm#internetCashManagement (the section labeled "Is Internet Cash Management secure"). The tokens are only offered to commercial users who opt in for the token and use them during login and Bill Pay. I will change the value to just be "hardware" instead, if that works.

**Comment by smmelot-16** [comment]
In regards to your first point, I think I can explain it a bit better:
1. Yes, the OTP and Quick Login methods mentioned are separate. The Quick Login feature is one customer's opt-in to and requires them to register a second device for authentication during login, hence the 2FA. The OTP method mentioned on the website is only used as a fallback, not as an option during authentication (essentially if they lost access to their 2nd device, they'd have to call in to speak to a rep to get the OTP sent).
2. I think I could probably add a note here, as Quick Login is offered for desktop logins, and not mobile app logins currently, so it's not a full 2FA solution, so to speak. The security questions are presented during mobile app logins where some risk-based rules triggered an alert, but aren't presented on desktop logins where the customer has verified their identity through their 2nd device.

**Comment by Carlgo11** [comment]
So if I understand you correctly, the options are:
- Personal
  - Website: 2FA through Push notifications (SMS OTP as backup)
  - App: No 2FA
- Business
  - Website: Some unknown hardware token if requested
  - App: N/A

**Comment by smmelot-16** [comment]
Yes, that is correct. If it's possible, could the term "Quick Login" be included in the description for the Software Token, or is it better to just leave that as is?

**Comment by Carlgo11** [comment]
I'm not sure it qualifies as 2FA if it can be bypassed. What do you think @phallobst?

**Comment by phallobst** [comment]
To be honest, the bank's documentation is severely lacking (to put it mildly), and confusing on top. On https://www.efirstbank.com/internet-banking/security-privacy.htm they say:

> You may also be asked to retrieve a verification code sent by email or SMS when you start using new products or services.

Is that still up-to-date, and if so, does it apply to logging in on a new computer (or after deleting their cookies)?

**Comment by Carlgo11** [comment]
As questions haven't been responded to yet I'm going to label this PR as inactive.
:warning: Unless there's activity within the next 7 days I'll close the PR.

**Comment by Carlgo11** [comment]
Closing due to inactivity.
> **Note**
> The PR author you can still commit the requested changes and re-open this PR.
> If someone else wants to commit the requested changes you can create a new PR and link to this PR.

**Pull request closed without merging**